libraries/AP_GPS/AP_GPS_SBP2.cpp

```diff
@@ -476,7 +481,35 @@ AP_GPS_SBP2::logging_log_raw_sbp(uint16_t msg_type,
             time_us         : time_us,
             msg_type        : msg_type,
         };
-        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,192));
+        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,104));
         gps._DataFlash->WriteBlock(&pkt2, sizeof(pkt2));
+
+        if (msg_len > 168) {
+            struct log_SbpRAW2 pkt3 = {
+                LOG_PACKET_HEADER_INIT(LOG_MSG_SBPRAW2),
```

Comment:
I'm not quite familiar enough with the dataflash threading, but I believe it's technically possible that the backend might drop the first message for a to full buffer, but accept the second one because it has since dumped the first one out. If this is possible then I suspect your log tool will be very unhappy. I'd support making a second identified even if the structure is otherwise identical.